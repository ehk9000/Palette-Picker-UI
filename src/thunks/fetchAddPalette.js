import { addPalette, setLoading, setError } from '../actions';

export const fetchAddPalette = (palette) => {
  return async (dispatch) => {
    const url ='http://localhost:3001/api/v1/palettes';
    const body = {
      color_1: palette.color_1,
      color_2: palette.color_2,
      color_3: palette.color_3,
      color_4: palette.color_4,
      color_5: palette.color_5,
      name: palette.name,
      project_name: palette.project_name
    }

    const options = {
      method: 'POST',
      headers: { 'Content-type': 'application/json' },
      body: JSON.stringify(body)
    }

    try {
      dispatch(setLoading(true));

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const palette = await response.json();

      dispatch(setLoading(false));
      dispatch(addPalette(palette));
    } catch(error) {
      dispatch(setError(error.message));
    }
  }
}