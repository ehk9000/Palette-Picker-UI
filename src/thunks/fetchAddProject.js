import { addProject, selectCurrentProject, setLoading, setError } from '../actions';

export const fetchAddProject = (project) => {
  return async (dispatch) => {
    const base = `https://color-me-newton-api.herokuapp.com`;
    const url = `${base}/api/v1/projects`;
    const body = JSON.stringify(project)
    const options = {
      method: 'POST',
      headers: { 'Content-type': 'application/json' },
      body
    }
    
    try {
      dispatch(setLoading(true));

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const project = await response.json();
      dispatch(setLoading(false));
      dispatch(addProject(project));
      dispatch(selectCurrentProject(project));
    } catch(error) {
      dispatch(setError(error.message));
    }
  }
}