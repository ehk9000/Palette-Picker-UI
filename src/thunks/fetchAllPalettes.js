import { addAllPalettes, setLoading, setError } from '../actions/index';

export const fetchAllPalettes = () => {
  return async (dispatch) => {
    const url = 'http://localhost:3000/api/v1/palettes';

    try {
      dispatch(setLoading(true));

      const response = await fetch(url);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const palettes = await response.json();

      dispatch(setLoading(false));
      dispatch(addAllPalettes(palettes));
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}