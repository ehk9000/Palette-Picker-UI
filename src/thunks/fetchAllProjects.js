import { addAllProjects, setLoading, setError } from '../actions/index';

export const fetchAllProjects = () => {
  return async (dispatch) => {
    const base = "https://color-me-newton-api.herokuapp.com";
    const url = `${base}/api/v1/projects`;

    try {
      dispatch(setLoading(true));

      const response = await fetch(url);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const projects = await response.json();


      dispatch(setLoading(false));
      dispatch(addAllProjects(projects));
    } catch(error) {
      dispatch(setError(error.message))
    }
  }
}