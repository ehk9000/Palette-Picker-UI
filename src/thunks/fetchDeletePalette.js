import { setLoading, setError, deletePalette, clearCurrentPalette } from '../actions';

export const fetchDeletePalette = (id) => {
  return async (dispatch) => {
    const options = {
      method: 'DELETE'
    }

    const url = `http://localhost:3000/api/v1/palettes/${id}`

    try {
      dispatch(setLoading(true))

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText)
      }

      dispatch(deletePalette(id));
      dispatch(clearCurrentPalette())
      dispatch(setLoading(false));
    } catch(error) {
      dispatch(setError(error.message))
    }
  }
}