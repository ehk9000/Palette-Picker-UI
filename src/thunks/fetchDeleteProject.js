import { setLoading, setError, deleteProject, clearCurrentProject } from '../actions';

export const fetchDeleteProject = (id) => {
  return async (dispatch) => {
    const options = {
      method: 'DELETE'
    }

    const base = "https://color-me-newton-api.herokuapp.com";
    const url = `${base}/api/v1/projects/${id}`;
    
    try {
      dispatch(setLoading(true))

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText)
      }

      dispatch(deleteProject(id));
      dispatch(clearCurrentProject())
      dispatch(setLoading(false));
    } catch(error) {
      dispatch(setError(error.message))
    }
  }
}