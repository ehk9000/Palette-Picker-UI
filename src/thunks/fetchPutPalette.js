import { updatePalette, setLoading, setError } from '../actions';

export const fetchPutPalette = (project) => {
  return async (dispatch) => {
    const url = `http://localhost:3000/api/v1/palettes/${palette.id}`;
    const body = JSON.stringify(palette)
    const options = {
      method: 'PUT',
      headers: { 'Content-type': 'application/json' },
      body
    }
    
    try {
      dispatch(setLoading(true));

      const response = await fetch(url, options);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const palette = await response.json();
      dispatch(setLoading(false));
      dispatch(updatePalette(palette));
    } catch(error) {
      dispatch(setError(error.message));
    }
  }
}