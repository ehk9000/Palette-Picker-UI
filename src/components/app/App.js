import React,{ Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchAllProjects } from '../../thunks/fetchAllProjects';
import { fetchAllPalettes } from '../../thunks/fetchAllPalettes';
import  Project_list from '../../containers/Project_list';
import ControlForm from '../../containers/control_form';
import Header from '../header';
require("dotenv").config();
process.env.

export class App extends Component {
  constructor() {
    super()
    this.state = {
      projects: [],
      palettes: []
    }
  }

  componentDidMount() {
    this.props.fetchAllProjects()
    this.props.fetchAllPalettes()
  }

  render() {
    return (
    <div className="App">
      <Header />
      <main>
        <Route exact path="/" component={ControlForm} />
        <Route exact path="/createNewProject" component={ControlForm} />
        <Route path="/projects" component={Project_list} />
      </main>
    </div>
  );

    // <Route path="/palettes" component={Palettes_list} />
  }
}

const mapStateToProps = ({projects, palettes}) => ({
  projects,
  palettes
});

const mapDispatchToProps = dispatch => ({
  fetchAllProjects: () => dispatch(fetchAllProjects()),
  fetchAllPalettes: () => dispatch(fetchAllPalettes()) 
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
