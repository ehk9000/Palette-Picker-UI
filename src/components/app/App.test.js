import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { fetchAllProjects } from '../../thunks/fetchAllProjects';
import { fetchAllPalettes } from '../../thunks/fetchAllPalettes';

jest.mock("../../thunks/fetchAllProjects");
jest.mock("../../thunks/fetchAllPalettes");


describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <App fetchAllPalettes={fetchAllPalettes} fetchAllProjects={fetchAllProjects}/> );
  });

  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe("componentDidMount", () => {
    fetchAllPalettes.mockImplementation(() => { });
    fetchAllProjects.mockImplementation(() => { });

    it("should call fetchAllPalettes", () => {
      expect(fetchAllPalettes).toHaveBeenCalled();
    });
    it("should call fetchAllProjects", () => {
      expect(fetchAllProjects).toHaveBeenCalled();
    });
  });
});

