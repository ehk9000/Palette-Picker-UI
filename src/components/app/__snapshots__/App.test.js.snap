// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correctly 1`] = `ShallowWrapper {}`;
